packages-tests/BetterPhpDocParser/PhpDocParser/ClassAnnotationMatcher/Fixture/fixture.php.inc

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Rector\Tests\BetterPhpDocParser\PhpDocParser\ClassAnnotationMatcher\Fixture;
+
+use Rector\Tests\BetterPhpDocParser\PhpDocParser\ClassAnnotationMatcher\Source\KnownClass;
+use My\Namespace\OtherUnknownClass;
+
+class MyClass
+{
+    /** @var KnownClass */
+    private $known;
+    /** @var \SplHeap */
+    private $knownGlobalClass;
+    /** @var UnknownClass */
+    private $unknownClass;
+    /** @var \My\Namespace\UnknownClass */
+    private $unknownNamespacedClass;
+//    /** @var OtherUnknownClass */
+//    private $unknownUsedClass;
```

Comment:
why it is commented?

Reply:
This is a testcase we currently cannot cover because the `OtherUnknownClass`  is imported (= used). So, this is a case we cannot detect properly. As far as I know, PHP ignores use statements pointing to non existant classes.
I could have removed this case, but I think it is worth mentioning. I'll write a helpful comment..

Reply:
How about create a test case for direct rector rule usage? eg, for `DoctrineTargetEntityStringToClassConstantRector` usage.

I think that will be clearer for the purpose and use case.

You can add the test case to `tests/Issues` directory as the rule is from rector-doctrine

Reply:
Well, the test is now specific for name resolution. I mean, yes, `DoctrineTargetEntityStringToClassConstantRector` will benefit from this feature, but other users of `ClassAnnotationMatcher->resolveTagFullyQualifiedName()` could benefit as well. So, I suggest to keep this test at a central place (here).

Reply:
I am not sure without real failing test case example in a rule, docblock is depends on the statement next to it, so it will be still read by next stmt (as property), possibly by children of  VarTagValueNode

Reply:
Hah. I hoped that it would be not be recognised as DocBlock..
Anyway. I have (hopefully) a better Idea: How about trying to fix this case as well with `class_exists()`? Then we could uncomment these lines..

Reply:
Could you add test for usage in rule, eg: on `DoctrineTargetEntityStringToClassConstantRector`? 

You can add the test case to tests/Issues directory as the rule is from rector-doctrine

Reply:
Ah. Now I understood what you meant in the first place. Apologies for getting you wrong and thanks for your patience. I added a Testcase under `tests/Issues/DoNotReplaceUnknownTargetEntities`. This test fails, because the DoctrineTargetEntityStringToClassConstantRector needs to use `ClassAnnotationMatcher->resolveTagToKnownFullyQualifiedName()` now.

Reply:
could you make test passed then?

Reply:
I made the tests reflect the status quo for now, so they pass. IMHO, it would be better to mark them as skipped, but I couldn't figure out how to do that.
I'll create other PRs in rector-doctrine to fix them properly.